packages/teleterm/src/ui/TabHost/TabHost.tsx

```diff
@@ -55,17 +55,25 @@ export default function TabHost(props: Props) {
     serviceDocs.openNewTerminal();
   }
 
+  const openTerminalContextMenu = () => {
+    mainProcessClient.openTerminalContextMenu();
+  };
+
   const $docs = documents.map(doc => {
     const isActiveDoc = doc === docActive;
-    return <MemoizedDocument doc={doc} visible={isActiveDoc} key={doc.uri} />;
+    return (
+      <MemoizedDocument
+        doc={doc}
+        visible={isActiveDoc}
+        key={doc.uri}
+        onTerminalContextMenu={openTerminalContextMenu}
```

Comment:
Do we need pass the `onTerminalContextMenu` from here down to Terminal? Can we reference the mainProcessCelint from the  `useDocumentTerminal` instead and clean up TabHost?

Reply:
No need to pass down the handleTabClose as the documents have access to `serviceDocs`

Reply:
Correct, removed handlers from `TabHost`